Update Ros2ManipulationTemplate

jhanca-robotecai opened the pull request:
## What does this PR do?

This PR updates `Update Ros2ManipulationTemplate` to make it fully functional. It fixes issues listed in #822 and more. It also introduces some of the changes suggested in the [RFC for simulation-related gems and templates](https://github.com/o3de/sig-simulation/pull/94)
- game launcher was set to start the default level
- obsolete replaces in the `RoboticManipulation` level were removed (the _prefabs_ were updated earlier)
- `PalletizationLevel` and all assets used only in this level were removed
- unused passes (causing problems in future 2505.0), unused baked diffuse probe grids and unused baked global illumination files were removed
- `Panda` robot was beautified (materials were added)
- `RoboticManipulation` level was modified: some materials were added to make it look better
- `.gitignore` was added to the template, so it can serve its purpose when created as a project
- the `template.json` file was regenerated to include all files and keep them in alphabetical order

## How was this PR tested?

`Ros2ManipulationTemplate` was built with the newest version of Gems and tested. It was built in `profile` with a game launcher and in `release monolithic`. Compiled against O3DE `9b8d468f64` _stabilization/25050_

The tests included:
- running Editor and checking for warnings/errors
- running GameLauncher and checking for warning/errors (the default level loads correctly)
- running `ros2 action` to move the arm for tests
- running `ros2 launchfile` to check if the camera image is visible, if the colored point cloud is visible if the moveit interface works correctly

The tests were conducted using ROS 2 Humble on Ubuntu 22.04. Tests on Jazzy were not performed.

Marking it as draft: it is built on top of #840

jhanca-robotecai commented:
This branch was rebased after #840 was merged; ready for review

pawelbudziszewski reviewed (changes requested):
Everything works as expected. Just a few comments:

- In the `Assets/Room/Textures` folder:
  - `Ambient` and `Height` textures (in all subfolders) are not used. Can be purged
  - Floor roughness texture (`plane_divided_DefaultMaterial_Roughness.png`) is not connected to the material (should be)
- The `Box toy puzzle` elements (cylinder, cube, and triangle) as well as the container don't have explicitly created material. In the default (implicit) material not all maps are connected
- In the folder with these textures (`Assets/BoxToyPuzzle/texture`):
  - `Glossiness`, `Height`, and `Emissive` are not used, can be purged
  - `Metallic` and `Specular` are plain, but I suggest to leave them (and connect to the material)
  - The file `Assets/BoxToyPuzzle/texture/Roughness/MCube_pazzle_Roughness.1001.png` seems to be an artifact - should be removed
  - It would be nice to clean file names (`rename 's/.1001//' *.png`)

pawelbudziszewski commented:
These are minor issues, but I think it would be nice to clean these textures while we're updating the template.

jhanca-robotecai commented:
> In the `Assets/Room/Textures` folder:
> * `Ambient` and `Height` textures (in all subfolders) are not used. Can be purged

I believe the two are used. Could you please confirm?

jhanca-robotecai commented:
>   * `Ambient` and `Height` textures (in all subfolders) are not used. Can be purged

I removed `Height` textures (`Ambient` textures are used)

>   * Floor roughness texture (`plane_divided_DefaultMaterial_Roughness.png`) is not connected to the material (should be)

Fixed

> * The `Box toy puzzle` elements (cylinder, cube, and triangle) as well as the container don't have explicitly created material. In the default (implicit) material not all maps are connected

It is connected now; obsolete textures are removed and `1001` is removed from the name. I did not change `pazzle`->`puzzle` typo.

>   * The file `Assets/BoxToyPuzzle/texture/Roughness/MCube_pazzle_Roughness.1001.png` seems to be an artifact - should be removed

Removed

pawelbudziszewski reviewed (approved):
LGTM
